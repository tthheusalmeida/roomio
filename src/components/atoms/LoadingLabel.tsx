import { mergeClassNames } from "@/utils/classNames";

interface LoadingLabelProps {
  label: string;
  className?: string;
}

export default function LoadingLabel({ label, className }: LoadingLabelProps) {
  return (
    <span className={mergeClassNames("animate-pulse", className)}>{label}</span>
  );
}
