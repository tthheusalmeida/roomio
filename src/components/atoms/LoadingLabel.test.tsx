import { render, screen } from "@testing-library/react";
import LoadingLabel from "./LoadingLabel";

describe("LoadingLabel component", () => {
  it("renders the label text with pulse animation", () => {
    const { asFragment } = render(<LoadingLabel label="Loading..." />);

    const span = screen.getByText("Loading...");
    expect(span).toBeInTheDocument();
    expect(span).toHaveClass("animate-pulse");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <LoadingLabel label="Loading..." className="extra-class" />
    );

    const span = screen.getByText("Loading...");
    expect(span).toHaveClass("animate-pulse", "extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
