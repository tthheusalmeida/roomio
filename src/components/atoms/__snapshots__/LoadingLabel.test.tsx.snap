// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingLabel component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <span
    class="animate-pulse extra-class"
  >
    Loading...
  </span>
</DocumentFragment>
`;

exports[`LoadingLabel component renders the label text with pulse animation 1`] = `
<DocumentFragment>
  <span
    class="animate-pulse"
  >
    Loading...
  </span>
</DocumentFragment>
`;
